Display `"0"` Value for Low Percentage Metrics in SuperAdmin Dashboard

saithsab877 opened the pull request:
### Problem:
In the metrics dashboard, particularly in the `"bounties -> completed"` and `"satoshis->paid"` sections, percentages close to 0 were not displayed, leading to a lack of clarity in the data presented. Users were unable to discern whether the lack of a value meant 0 or an error in data retrieval.

### Expected Behavior:
The dashboard should display a `"0"` value for any percentage that is below `0.00%` in the `"bounties -> completed"` and `"satoshis -> paid"` sections to improve data clarity and user understanding of the dashboard metrics.

## Issue ticket number and link:
- **Ticket Number:** [ 208 ]
- **Link:** [ https://github.com/stakwork/sphinx-tribes-frontend/issues/208 ]

### Solution:
To resolve this issue, a utility function `formatPercentage` was introduced to format percentage values. This function ensures that any percentage value below 0.01 (and greater than 0) is displayed as `"0"`. This logic was applied to the metrics display in the Statistics component, specifically for `bounties_paid_average` and `sats_paid_percentage`.

### Changes:
- Added `formatPercentage` function to format percentage values correctly.
- Updated the `Statistics` component to utilize `formatPercentage` for displaying percentage values in the `"bounties -> completed"` and `"satoshis -> paid"` sections.
- Ensured consistency in displaying "0" for low percentage values across different sections of the metrics dashboard.

### Evidence:
 Please see the attached video as evidence.
- Demo: [Link ](https://www.loom.com/share/5887c6728c1242af9dc4c04304e5e7d5)

### Testing:

**1. Browser Compatibility:**
- Tested the feature extensively on Chrome to ensure consistent behavior.

**2. Unit Testing:**
- Unit tests were written and updated to cover the new `formatPercentage` function, ensuring it correctly formats the values as expected across a range of test cases.
- Additional tests were conducted to confirm that the Statistics component correctly applies the formatting to percentage values before rendering, ensuring robustness and reliability of the display logic.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome
- [x] I have provided a recording.

saithsab877 commented:
Hi @ecurrencyhodler, @elraphty, @kevkevinpal
Please review this PR

aliraza556 commented:
Thank you, @saithsab877! Looks good to me. Additionally, here's a neat tip: if you paste the Loom video link by itself, it embeds the entire video directly into the PR! Let's proceed with the review.

aliraza556 commented:
I have examined the reported issue and the associated code modifications. Additionally, I carried out local testing to verify that everything operates correctly and does not impact any other functionalities.

ecurrencyhodler commented:
Video LGTM.

elraphty commented on src/pages/superadmin/statistics/index.tsx:
```diff
@@ -36,6 +36,16 @@ interface StatisticsProps {
   mockHunter?: MockHunterMetrics;
 }
 
+const formatPercentage = (value?: number): string => {
```
Please add this to helpers, and write a unit test for it. src/helpers/helpers-extended.ts

saithsab877 replied:
Absolutely, I'll add the requested changes to helpers-extended.ts and ensure a unit test is written for it. I'll let you know as soon as it's completed.

saithsab877 commented:
Hi @elraphty,
Please check it out. I have fixed it and also added a unit test for both the `src/pages/superadmin/statistics/__tests__/Statistics.spec.tsx` and the `src/helpers/__test__/helpers.spec.tsx files.`

kevkevinpal commented:
the change looks good to me thanks for the detailed description! also nice work on the testing

Will merge once the CI is finished running

kevkevinpal commented:
fixed merge conflict and force pushed to  0e0de8c

ecurrencyhodler commented:
Sorry @saithsab877 I did see one inconsistency in staging. I will you you out right now but can you do a fast follow?

Here I see 4.00% for bounties and 0% for sats. These should be the same format. Let's keep both at x% instead of x.xx%
<img width="784" alt="Screenshot 2024-02-09 at 10 26 18 AM" src="https://github.com/stakwork/sphinx-tribes-frontend/assets/32662508/e4f44c3c-2a89-481f-bf9c-9a8194ec91cb">

saithsab877 commented:
> Sorry @saithsab877 I did see one inconsistency in staging. I will you you out right now but can you do a fast follow?
> 
> Here I see 4.00% for bounties and 0% for sats. These should be the same format. Let's keep both at x% instead of x.xx% <img alt="Screenshot 2024-02-09 at 10 26 18 AM" width="784" src="https://private-user-images.githubusercontent.com/32662508/303742212-e4f44c3c-2a89-481f-bf9c-9a8194ec91cb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc1MDM2MTgsIm5iZiI6MTcwNzUwMzMxOCwicGF0aCI6Ii8zMjY2MjUwOC8zMDM3NDIyMTItZTRmNDRjM2MtMmE4OS00ODFmLWJmOWMtOWE4MTk0ZWM5MWNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA5VDE4MjgzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUyMmQ2MzFjNDRmZmM2NTY0NGU0MjBjZjlkM2FlMTE1ZjUwY2VmNGNlN2U4OGQ1ZTc0YTBkMGI3MzIzZDIxYTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KqLmb0bXhCf0mGzptDv61AheapNHUcmcr3wpiL9-Jtc">

Thanks for catching that, @ecurrencyhodler! I'll update the formats to 'x%' for both bounties and sats right away. Will follow up soon with the changes.

ecurrencyhodler commented:
Thanks! and your bounty has been paid. :)

saithsab877 commented:
Hi @ecurrencyhodler,
Please check it https://www.loom.com/share/1e3c3e03e552425cbb06f9aecdc53ea4

ecurrencyhodler commented:
Okay that looks great ty.

saithsab877 commented:
Hi @ecurrencyhodler
I create a new PR?

ecurrencyhodler commented:
Yes that would be great.
